### PR TITLE
Module 1: Add a web application docker

### PR DESCRIPTION
**Pull request by angeliney**

### CODE REVIEW

File: module-1/app-web/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+FROM python:3.8
+
+# set a directory for the app
+WORKDIR /usr/src/app
+
+# copy all the files to the container
+COPY . .
+
+# install dependencies
+RUN pip install --no-cache-dir -r requirements.txt
```
Comment (truskovskiyk) [inline module-1/app-web/Dockerfile]: best practice is 

1. to copy `requirements.txt`
2. install them: pip install --no-cache-dir -r requirements.txt
3. copy rest of the code

Why? Docker uses cache a lot, and usually installing requirements.txt takes time, but requirements.txt does not change as often as the code. So every time you copy everything and install requirements.txt, you install them from scratch. In the case above, you just rely on the cache.

File: module-1/app-web/app.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    "https://firebasestorage.googleapis.com/v0/b/docker-curriculum.appspot.com/o/catnip%2F2.gif?alt=media&token=8a108bd4-8dfc-4dbc-9b8c-0db0e626f65b",
+    "https://firebasestorage.googleapis.com/v0/b/docker-curriculum.appspot.com/o/catnip%2F3.gif?alt=media&token=4e270d85-0be3-4048-99bd-696ece8070ea",
+    "https://firebasestorage.googleapis.com/v0/b/docker-curriculum.appspot.com/o/catnip%2F4.gif?alt=media&token=e7daf297-e615-4dfc-aa19-bee959204774",
+    "https://firebasestorage.googleapis.com/v0/b/docker-curriculum.appspot.com/o/catnip%2F5.gif?alt=media&token=a8e472e6-94da-45f9-aab8-d51ec499e5ed",
+    "https://firebasestorage.googleapis.com/v0/b/docker-curriculum.appspot.com/o/catnip%2F7.gif?alt=media&token=9e449089-9f94-4002-a92a-3e44c6bd18a9",
+    "https://firebasestorage.googleapis.com/v0/b/docker-curriculum.appspot.com/o/catnip%2F8.gif?alt=media&token=80a48714-7aaa-45fa-a36b-a7653dc3292b",
+    "https://firebasestorage.googleapis.com/v0/b/docker-curriculum.appspot.com/o/catnip%2F9.gif?alt=media&token=a57a1c71-a8af-4170-8fee-bfe11809f0b3",
+]
```
Comment (truskovskiyk) [inline module-1/app-web/app.py]: optional: I would move this to JSON or something. or define `https://firebasestorage.googleapis.com/v0/b/docker-curriculum.appspot.com/o/`once

File: module-1/app-web/app.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+
+@app.route("/")
+def index():
+    url = random.choice(images)
+    return render_template("index.html", url=url)
```
Comment (truskovskiyk) [inline module-1/app-web/app.py]: nice! you are full-stack developer now!

File: module-1/app-web/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+Flask==2.0.2
```
Comment (truskovskiyk) [inline module-1/app-web/requirements.txt]: fast api is much more modern & popular right now

File: module-1/app-web/templates/index.html [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      }
+      div.container {
+        max-width: 500px;
+        margin: 100px auto;
+        border: 20px solid white;
+        padding: 10px;
+        text-align: center;
+      }
+      h4 {
+        text-transform: uppercase;
```
Comment (truskovskiyk) [inline module-1/app-web/templates/index.html:16]: image

File: module-1/kubernetes-app/deployment-app-web.yaml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    matchLabels:
+      app: deployments-app-web
+  template:
+    metadata:
+      labels:
+        app: deployments-app-web
+    spec:
+      containers:
+        - name: app-web
+          image: hub.docker.com/r/angelineygeorgian/catnip:latest
```
Comment (truskovskiyk) [inline module-1/kubernetes-app/deployment-app-web.yaml]: is this docker hub image? could be just `angelineygeorgian/catnip:latest`

File: module-1/kubernetes-app/job-app-hello-world.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: job-app-hello-world
+spec:
+  parallelism: 2
+  template:
+    spec:
+      containers:
+        - image: hub.docker.com/r/angelineygeorgian/hello-world/:latest
```
Comment (truskovskiyk) [inline module-1/kubernetes-app/job-app-hello-world.yaml]: same, why do you need `hub.docker.com/r/`

File: module-1/kubernetes-app/pod-app-hello-world.yaml [hunk 8]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod-app-hello-world
+spec:
+  containers:
+    - image: hub.docker.com/r/angelineygeorgian/hello-world:latest
```
Comment (truskovskiyk) [inline module-1/kubernetes-app/pod-app-hello-world.yaml]: same question, remove`hub.docker.com/r/`

File: module-1/kubernetes-app/pod-app-web.yaml [hunk 9]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod-app-web
+spec:
+  containers:
+    - image: hub.docker.com/r/angelineygeorgian/catnip:latest
```
Comment (truskovskiyk) [inline module-1/kubernetes-app/pod-app-web.yaml]: same. do you need `hub.docker.com/r/`

File: module-1/kubernetes-app/deployment-app-web.yaml (hunk 6 above)
Comment (truskovskiyk) [inline module-1/kubernetes-app/deployment-app-web.yaml]:
```
docker run -it angelineygeorgian/catnip:latest /bin/bash
Unable to find image 'angelineygeorgian/catnip:latest' locally
latest: Pulling from angelineygeorgian/catnip
```

could you share docker image with me?

**Review by truskovskiyk — APPROVED** [review]
@angeliney great job!

**Comment by truskovskiyk** [comment]
@angeliney feel free to merge this

File: module-1/app-web/app.py (hunk 3 above)
Comment (angeliney) [reply module-1/app-web/app.py]: haha thank you! I copied it from docker curriculum mostly, but I've updated this to fastAPI now

**Pull request merged**